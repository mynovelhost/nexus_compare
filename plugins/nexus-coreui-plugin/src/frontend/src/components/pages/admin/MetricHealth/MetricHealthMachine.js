/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {assign, Machine} from 'xstate';
import Axios from 'axios';

export const ASC = 1;
export const DESC = -1;

export default Machine(
    {
      initial: 'loading',

      context: {
        data: [],
        pristineData: [],
        sortField: 'name',
        sortDirection: ASC,
        error: ''
      },

      states: {
        loading: {
          invoke: {
            src: 'fetchMetricHealth',
            onDone: {
              target: 'loaded',
              actions: ['setData']
            },
            onError: {
              target: 'error',
              actions: ['setError']
            }
          }
        },
        loaded: {
          entry: ['sortData'],
          on: {
            SORT_BY_NAME: {
              target: 'loaded',
              actions: ['setSortByName']
            },
            SORT_BY_MESSAGE: {
              target: 'loaded',
              actions: ['setSortByMessage']
            },
            SORT_BY_ERROR: {
              target: 'loaded',
              actions: ['setSortByError']
            }
          }
        },
        error: {}
      }
    },
    {
      actions: {
        setData: assign({
          data: (_, event) => Object.entries(event.data.data).map(metric => Object.assign({name: metric[0]}, metric[1])),
          pristineData: (_, event) => Object.entries(event.data.data).map(metric => Object.assign({name: metric[0]}, metric[1]))
        }),
        setError: assign({
          error: (_, event) => event.data.message
        }),
        setSortByName: assign({
          sortField: 'name',
          sortDirection: ({sortField, sortDirection}) => {
            if (sortField !== 'name') {
              return ASC;
            }
            return sortDirection === ASC ? DESC : ASC
          }
        }),
        setSortByMessage: assign({
          sortField: 'message',
          sortDirection: ({sortField, sortDirection}) => {
            if (sortField !== 'message') {
              return ASC;
            }
            return sortDirection === ASC ? DESC : ASC
          }
        }),
        setSortByError: assign({
          sortField: 'error',
          sortDirection: ({sortField, sortDirection}) => {
            if (sortField !== 'error') {
              return ASC;
            }
            return sortDirection === ASC ? DESC : ASC
          }
        }),
        sortData: assign({
          data: ({sortField, sortDirection, pristineData}) => (pristineData.slice().sort((a, b) => {
            return a[sortField] > b[sortField] ? sortDirection : -sortDirection;
          }))
        })
      },
      services: {
        fetchMetricHealth: () => Axios.get('/service/rest/internal/ui/status-check')
      }
    }
);
