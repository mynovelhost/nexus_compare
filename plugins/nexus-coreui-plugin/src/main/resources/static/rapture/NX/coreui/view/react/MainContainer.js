/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
