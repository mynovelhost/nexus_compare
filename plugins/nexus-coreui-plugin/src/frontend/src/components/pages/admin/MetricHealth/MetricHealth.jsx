/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {useMachine} from '@xstate/react';

import './MetricHealth.scss';

import {faCheckCircle,faExclamationCircle} from '@fortawesome/free-solid-svg-icons';

import {
  Alert,
  ContentBody,
  Button,
  NxFontAwesomeIcon,
  NxFilterInput,
  NxTable,
  NxTableBody,
  NxTableCell,
  NxTableHead,
  NxTableRow,
} from 'nexus-ui-plugin';

import MetricHealthMachine, {ASC} from './MetricHealthMachine';

import UIStrings from "../../../../constants/UIStrings";

export default function MetricHealth() {
  const [current, send] = useMachine(MetricHealthMachine, {devTools: true});
  const isLoading = current.matches('loading');
  const data = current.context.data;
  const error = current.context.error;

  const nameSortDir = getSortDirection('name', current.context.sortField, current.context.sortDirection);
  const messageSortDir = getSortDirection('message', current.context.sortField, current.context.sortDirection);
  const errorSortDir = getSortDirection('error', current.context.sortField, current.context.sortDirection);

  function getSortDirection(fieldName, sortField, sortDirection) {
    if (sortField !== fieldName) {
      return null;
    }
    return sortDirection === ASC ? 'asc' : 'desc';
  }

  return (
    <ContentBody className="nxrm-metric-health">
      <NxTable>
        <NxTableHead>
          <NxTableRow>
            <NxTableCell hasIcon />
            <NxTableCell onClick={() => send('SORT_BY_NAME')} isSortable sortDir={nameSortDir}>
              {UIStrings.METRIC_HEALTH.NAME_HEADER}
            </NxTableCell>
            <NxTableCell onClick={() => send('SORT_BY_MESSAGE')} isSortable sortDir={messageSortDir}
              >{UIStrings.METRIC_HEALTH.MESSAGE_HEADER}
            </NxTableCell>
            <NxTableCell onClick={() => send('SORT_BY_ERROR')} isSortable sortDir={errorSortDir}>
              {UIStrings.METRIC_HEALTH.ERROR_HEADER}
            </NxTableCell>
          </NxTableRow>
        </NxTableHead>
        <NxTableBody isLoading={isLoading} error={error}>
          {data.map(metric => (
            <NxTableRow key={metric.name}>
              <NxTableCell hasIcon>
                <NxFontAwesomeIcon
                  className={metric.healthy ? 'nxrm-metric-health--healthy' : 'nxrm-metric-health--unhealthy'}
                  icon={metric.healthy ? faCheckCircle : faExclamationCircle}/>
              </NxTableCell>
              <NxTableCell>{metric.name}</NxTableCell>
              <NxTableCell><span dangerouslySetInnerHTML={{__html: metric.message}} /></NxTableCell>
              <NxTableCell>{metric.error ? metric.error.message : ''}</NxTableCell>
            </NxTableRow>
          ))}
        </NxTableBody>
      </NxTable>
    </ContentBody>
  );
}
