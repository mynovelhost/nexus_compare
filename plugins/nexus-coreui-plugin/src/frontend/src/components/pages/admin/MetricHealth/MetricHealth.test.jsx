/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import { act } from 'react-dom/test-utils';
import {render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import axios from 'axios';

import MetricHealth from './MetricHealth';

jest.mock('axios', () => ({
  ...jest.requireActual('axios'), // Use most functions from actual axios
  get: jest.fn()
}));

describe('MetricHealth', function() {
  const renderView = async () => {
    var selectors;

    await act(async () => {
      const {container, queryByText} = render(<MetricHealth/>);

      selectors = {
        container,
        loadingMask: () => queryByText("Loadingâ€¦"),
      }
    });

    return selectors;
  };

  it('renders the resolved data', async function() {
    const metrics = {
      "Available CPUs" : {
        "healthy" : true,
        "message" : "The host system is allocating a maximum of 6 cores to the application.",
        "error" : null,
        "details" : null,
        "time" : 1588629343436,
        "duration" : 0,
        "timestamp" : "2020-05-04T15:55:43.436-06:00"
      },
      "testError" : {
        "healthy" : false,
        "message" : "TestError",
        "error" : {
          "cause" : null,
          "stackTrace" : [ {
            "methodName" : "getSystemStatusChecks",
            "fileName" : "HealthCheckResource.java",
            "lineNumber" : 61,
            "className" : "org.sonatype.nexus.internal.rest.HealthCheckResource",
            "nativeMethod" : false
          }, {
            "methodName" : "CGLIB$getSystemStatusChecks$0",
            "fileName" : "<generated>",
            "lineNumber" : -1,
            "className" : "org.sonatype.nexus.internal.rest.HealthCheckResource$$EnhancerByGuice$$1b6f529d",
            "nativeMethod" : false
          } ],
          "message" : "TestError",
          "localizedMessage" : "TestError",
          "suppressed" : [ ]
        }
      }
    };

    axios.get.mockReturnValue(Promise.resolve({
      data: metrics
    }));

    const {container, loadingMask} = await renderView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(container.querySelector(`tbody tr:nth-child(1) td:nth-child(2)`)).toHaveTextContent("Available CPUs");
    expect(container.querySelector(`tbody tr:nth-child(1) td:nth-child(3)`)).toHaveTextContent("The host system is allocating a maximum of 6 cores to the application.");

    expect(container).toMatchSnapshot();
  });

  it('renders a loading spinner', async function() {
    axios.get.mockReturnValue(new Promise(() => {}));

    const {container, loadingMask} = await renderView();

    expect(loadingMask()).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders an error message', async function() {
    axios.get.mockReturnValue(Promise.reject({message: 'Error'}));

    const {container, loadingMask} = await renderView();

    await wait(() => expect(loadingMask()).not.toBeInTheDocument());

    expect(container.querySelector('td.nx-error')).toHaveTextContent('Error');
    expect(container).toMatchSnapshot();
  });
});
