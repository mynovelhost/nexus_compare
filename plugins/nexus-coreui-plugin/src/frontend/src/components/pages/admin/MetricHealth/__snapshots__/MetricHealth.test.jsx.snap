// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricHealth renders a loading spinner 1`] = `
<div>
  <div
    class="nxrm-content-body nxrm-metric-health"
  >
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Message
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Error
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-cell--empty"
            colspan="4"
          >
            <div
              class="nx-loading-spinner"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-notch fa-w-16 fa-spin nx-loading-spinner__icon nx-icon"
                data-icon="circle-notch"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Loadingâ€¦
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`MetricHealth renders an error message 1`] = `
<div>
  <div
    class="nxrm-content-body nxrm-metric-health"
  >
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Message
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Error
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-error"
            colspan="4"
          >
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-fw nx-icon"
                data-icon="exclamation-triangle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                  fill="currentColor"
                />
              </svg>
              Error
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`MetricHealth renders the resolved data 1`] = `
<div>
  <div
    class="nxrm-content-body nxrm-metric-health"
  >
    <table
      class="nx-table"
    >
      <thead>
        <tr
          class="nx-table-row nx-table-row--header"
        >
          <th
            class="nx-cell nx-cell--header nx-cell--icon"
          />
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Name
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-down fa-w-10 nx-icon"
                data-icon="sort-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort-up fa-w-10 nx-icon"
                data-icon="sort-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Message
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
          <th
            class="nx-cell nx-cell--header nx-cell--sortable"
          >
            <span>
              Error
            </span>
            <span
              class="nx-cell__sort-icons fa-layers fa-fw"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sort fa-w-10 nx-icon"
                data-icon="sort"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-cell--icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check-circle fa-w-16 nxrm-metric-health--healthy nx-icon"
              data-icon="check-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                fill="currentColor"
              />
            </svg>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              Available CPUs
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              The host system is allocating a maximum of 6 cores to the application.
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span />
          </td>
        </tr>
        <tr
          class="nx-table-row"
        >
          <td
            class="nx-cell nx-cell--icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-circle fa-w-16 nxrm-metric-health--unhealthy nx-icon"
              data-icon="exclamation-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              testError
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              TestError
            </span>
          </td>
          <td
            class="nx-cell"
          >
            <span>
              TestError
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
